### PR TITLE
hostip: allow debug builds to override what host name to resolve

### PR DESCRIPTION
**Pull request by bagder**
The environment variable CURL_URL_HOSTNAME can be set to a host name
that will make libcurl resolve and use that host name instead of the
name it would otherwise use. Only for debug builds.

This also removes the special case debug support for LocalHost in the
interface string that kind of provided a similar debug approach. That
was brought in 7e07da977ce748 but doesn't seem to be used these days.

This is offered as an alternative take on @cmeister2's #1957.

### CODE REVIEW

File: lib/hostip.c [hunk 1]
```diff
@@ -450,6 +446,11 @@ int Curl_resolv(struct connectdata *conn,
   struct Curl_easy *data = conn->data;
   CURLcode result;
   int rc = CURLRESOLV_ERROR; /* default to failure */
+#ifdef DEBUGBUILD
+  char *fakehost = curl_getenv("CURL_URL_HOSTNAME");
```
Comment (cmeister2) [inline lib/hostip.c]: If you could change this line to:
```
#ifndef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION
char *fakehost = "127.0.0.1"
#else
char *fakehost = curl_getenv("CURL_URL_HOSTNAME");
#endif
```
then we still have the debug mode behaviour but the fuzzing code doesn't depend on an environment variable.

**Comment by cmeister2** [comment]
Alternatively how do you feel about adding a new CURLOPT which forces all hostname lookups to use the specified hostname instead? Would take a string argument and do pretty much exactly what your code does now.

**Comment by bagder** [comment]
That sounds very similar to [CURLOPT_CONNECT_TO](https://curl.haxx.se/libcurl/c/CURLOPT_CONNECT_TO.html) but supporting a wildcard for the HOST part? and probably for the PORT part as well.

**Comment by bagder** [comment]
Oh, that option already supports wildcards =)

**Comment by cmeister2** [comment]
Will have to see if that suffices :smile:

**Comment by cmeister2** [comment]
Yep, looks to be fine. You can probably close this then!

**Comment by bagder** [comment]
Cool, thanks @cmeister2 !

**Pull request closed without merging**